### PR TITLE
Nunit selenium

### PR DESCRIPTION
**Pull request by XuluWarrior**

### CODE REVIEW

File: openfin-csharp-api-test/openfin-csharp-api-test/OpenfinTests.cs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+        HttpFileServer fileServer;
+
+        Runtime runtime;
+
+        [SetUp]
+        public void SetUp()
+        {
+            string dir = Path.GetDirectoryName(GetType().Assembly.Location);
+            string dirToServe = Path.Combine(dir, "../../../../src");
+            fileServer = new HttpFileServer(dirToServe, 9070);
```
Comment (mwhitworth) [inline openfin-csharp-api-test/openfin-csharp-api-test/OpenfinTests.cs]: perhaps a comment as to why we need our own `HttpFileServer`?

File: openfin-csharp-api-test/openfin-csharp-api-test/Program.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
+﻿using Openfin.Desktop;
```
Comment (mwhitworth) [inline openfin-csharp-api-test/openfin-csharp-api-test/Program.cs]: How is this file used?

File: openfin-csharp-api-test/openfin-csharp-api-test/OpenfinTests.cs [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+            }
+            driver = null;
+        }
+
+        [TearDown]
+        public async Task TearDown()
+        {
+            if (fileServer != null)
+            {
+                fileServer.Stop();
```
Comment (mwhitworth) [inline openfin-csharp-api-test/openfin-csharp-api-test/OpenfinTests.cs]: `fileServer?.Stop()`

File: openfin-csharp-api-test/openfin-csharp-api-test/OpenfinTests.cs [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+            string appConfigArg = "--config=http://localhost:9070/app.json";
+            if (runtime != null)
+            {
+                options.DebuggerAddress = "localhost:4444";
+                Process.Start(runOpenfinPath, appConfigArg);
+            } else
+            {
+                options.BinaryLocation = runOpenfinPath;
+                options.AddArgument(appConfigArg);
+                options.AddArgument("--remote-debugging-port=4444");
```
Comment (mwhitworth) [inline openfin-csharp-api-test/openfin-csharp-api-test/OpenfinTests.cs]: Factor out `4444` and `9070` as constants?

File: openfin-csharp-api-test/openfin-csharp-api-test/OpenfinTests.cs [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+            }
+            driver = new ChromeDriver(service, options);
+        }
+
+        private Task<Runtime> ConnectToRuntime()
+        {
+            var taskCompletionSource = new TaskCompletionSource<Runtime>();
+
+            RuntimeOptions options = new RuntimeOptions();
+            options.Version = "14.78.46.23";
```
Comment (mwhitworth) [inline openfin-csharp-api-test/openfin-csharp-api-test/OpenfinTests.cs]: factor out as constant?

File: openfin-csharp-api-test/openfin-csharp-api-test/OpenfinTests.cs [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+                taskCompletionSource.SetResult(isRunning);
+            }, (Ack ack) =>
+            {
+                // Error
+            });
+            return taskCompletionSource.Task;
+        }
+
+        [Test]
+        public void Test()
```
Comment (mwhitworth) [inline openfin-csharp-api-test/openfin-csharp-api-test/OpenfinTests.cs]: Looks like something we can remove :)

**Review by mwhitworth — CHANGES_REQUESTED** [review]
Perhaps worth linking to https://github.com/openfin/hello-openfin-selenium-example somewhere? (I presume this is where the code was adapted from)

I think the coverage is good for now though - given the main priority is to check basic runtime compatibility

File: openfin-csharp-api-test/openfin-csharp-api-test/Program.cs (hunk 2 above)
Comment (XuluWarrior) [reply openfin-csharp-api-test/openfin-csharp-api-test/Program.cs]: I've removed the class methods.  But the tests only seem to run when project is a Console Application.  So the `Main` method is required.

File: README.md [hunk 7]
```diff
@@ -1,216 +1,13 @@
...
-However, if you run this code, it may call on the OpenFin RVM or OpenFin Runtime, which are covered by OpenFin’s Developer, Community, and Enterprise licenses. You can learn more about OpenFin licensing at the links listed below or just email us at support@openfin.co with questions.
-
-https://openfin.co/developer-agreement/ <br/>
-https://openfin.co/licensing/
-
-## Support
-Please enter an issue in the repo for any questions or problems. 
-<br> Alternatively, please contact us at support@openfin.co
+## Limitations
+Currently each nunit needs to run independently to ensure they run successfully.
```
Comment (mwhitworth) [inline README.md]: each nunit test?

**Review by mwhitworth — APPROVED** [review]

**Pull request merged**